src/priceHistory/saga.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Price,
+  fetchPriceHistoryFailure,
+  fetchPriceHistoryStart,
+  fetchPriceHistorySuccess,
+} from 'src/priceHistory/slice'
+import Logger from 'src/utils/Logger'
+import { ensureError } from 'src/utils/ensureError'
+import { fetchWithTimeout } from 'src/utils/fetchWithTimeout'
+import { safely } from 'src/utils/safely'
+import networkConfig from 'src/web3/networkConfig'
+import { call, put, spawn, takeLatest } from 'typed-redux-saga'
+
+const TAG = 'priceHistory/saga'
+
+/** @beta - Exclude from Knip dep check */
+export async function fetchTokenPriceHistory(
+  tokenId: string,
+  startTimestamp?: number,
+  endTimestamp?: number
+): Promise<Price[]> {
+  const queryParams = new URLSearchParams()
+  if (startTimestamp) {
+    queryParams.append('startTimestamp', `${startTimestamp}`)
+  }
+  if (endTimestamp) {
+    queryParams.append('endTimestamp', `${endTimestamp}`)
+  }
+
+  const url = `${
+    networkConfig.blockchainApiUrl
+  }/tokensInfo/${tokenId}/priceHistory?${queryParams.toString()}`
+  const response = await fetchWithTimeout(url)
+  if (!response.ok) {
+    throw new Error(
+      `Failed to fetch price history for ${tokenId}: ${response.status} ${response.statusText}`
+    )
+  }
+  return await response.json()
+}
+
+export function* fetchTokenPriceHistorySaga({
+  payload: { tokenId, startTimestamp, endTimestamp },
+}: ReturnType<typeof fetchPriceHistoryStart>) {
+  try {
+    if (!tokenId) {
+      throw new Error('TokenId is required')
+    }
```

Comment:
do we need this? tokenId is typed as string, not string | undefined

Reply:
Removed this check in 4b2c2c81fa9ac4d36af3849c947f86b4138da9cf!